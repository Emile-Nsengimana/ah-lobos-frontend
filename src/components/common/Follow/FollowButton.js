/* eslint-disable consistent-return */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import { Link } from "react-router-dom";
import followImage from "../../../assets/images/follow.svg";
import unFollowImage from "../../../assets/images/unfollow.svg";
import {
  followUser,
  unFollowUser,
  getFollowee
} from "../../../redux/actions/followAction";

export class FollowButton extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.onClick = this.onClick.bind(this);
  }

  async onClick() {
    const token = sessionStorage.getItem("token");
    const msg = document.getElementById("warn-follow");
    if (!token) {
      msg.classList.add("show-it");
    }
    await this.props.followUser(token, this.props.author);
  }

  async unFollowonClick() {
    const token = sessionStorage.getItem("token");
    await this.props.unFollowUser(token, this.props.author);
  }

  render() {
    return (
      <>
        <button className="follow">
          {this.props.isFollowing ? (
            <img src={followImage} alt="" onClick={this.onClick} />
          ) : (
            <img src={unFollowImage} alt="" onClick={this.unFollowonClick} />
          )}
        </button>
        <div id="warn-follow" className="login-to-follow">
          Please{" "}
          <Link to={`/login?redirect=/article/${this.props.slug}`}> login</Link>
          to follow this user
        </div>
      </>
    );
  }
}
const mapStateToProps = state => ({
  follow: state.followAuser,
  unfollow: state.unFollowAuser,
  isFollowing: state.isFollowing
});

FollowButton.propTypes = {
  followUser: propTypes.func.isRequired,
  unFollowUser: propTypes.func.isRequired,
  isFollowing: propTypes.bool.isRequired,
  slug: propTypes.string,
  author: propTypes.string
};
export default connect(
  mapStateToProps,
  { followUser, unFollowUser, getFollowee }
)(FollowButton);
