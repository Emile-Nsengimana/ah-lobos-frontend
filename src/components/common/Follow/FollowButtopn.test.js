import React from "react";
import { shallow } from "enzyme";
import { FollowButton } from "./FollowButton";

const props = {
  followee: "",
  followUser: jest.fn(),
  unFollowUser: jest.fn()
};

describe("get followees component", () => {
  const component = shallow(<FollowButton {...props} />);
  it("renders without errors", () => {
    expect(component).toMatchSnapshot();
  });
  it("should check `onClick()`", () => {
    const instance = component.instance();
    jest.spyOn(instance, "onClick");
    instance.onClick();
    expect(instance.onClick).toHaveBeenCalledTimes(1);
  });
  it("should check `unFollowonClick()`", () => {
    const instance = component.instance();
    jest.spyOn(instance, "unFollowonClick");
    instance.unFollowonClick();
    expect(instance.unFollowonClick).toHaveBeenCalledTimes(1);
  });
});
