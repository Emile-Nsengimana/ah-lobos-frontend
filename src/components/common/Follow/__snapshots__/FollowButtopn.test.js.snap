// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get followees component renders without errors 1`] = `ShallowWrapper {}`;
